From 51f25d7dfe770937041c07bdf05f979e6d73f007 Mon Sep 17 00:00:00 2001
From: Rene Engelhard <rene@debian.org>
Date: Mon, 18 Sep 2017 11:43:51 +0200
Subject: [PATCH] use pkg-config --modversion to check for ICU version

... since icu-config is removed at least in Debians icu 59.1 packages.
icu-i18n exists since exactly 4.6 (which we require anyway) so we can
remove that check, too

Change-Id: I28bd1509dcc2c52e22f91fb26421fcd585323b24
---
 configure.ac | 15 +++------------
 1 file changed, 3 insertions(+), 12 deletions(-)

diff --git a/configure.ac b/configure.ac
index f053df0192c7..0e02743e0ac3 100644
--- a/configure.ac
+++ b/configure.ac
@@ -8955,23 +8955,14 @@ if test "$with_system_icu" = "yes"; then
     AC_LANG_POP([C++])
 
     if test "$cross_compiling" != "yes"; then
-        AC_PATH_PROG(ICUCONFIG,icu-config)
-
-        AC_MSG_CHECKING([ICU version])
-        ICU_VERSION=`$ICUCONFIG --noverify --version 2>/dev/null || $ICUCONFIG --version`
+        PKG_CHECK_MODULES(ICU, icu-i18n >= 4.6)
+        ICU_VERSION=`$PKG_CONFIG --modversion icu-i18n 2>/dev/null`
         ICU_MAJOR=`echo $ICU_VERSION | cut -d"." -f1`
         ICU_MINOR=`echo $ICU_VERSION | cut -d"." -f2`
-
-        if test "$ICU_MAJOR" -ge "49" -o \( "$ICU_MAJOR" = "4" -a "$ICU_MINOR" -ge "6" \); then
-            AC_MSG_RESULT([OK, $ICU_VERSION])
-        else
-            AC_MSG_ERROR([not suitable, only >= 4.6 supported currently])
-        fi
     fi
 
     if test "$cross_compiling" = "yes" -a \( "$with_system_icu_for_build" = "yes" -o "$with_system_icu_for_build" = "force" \); then
-        AC_PATH_PROG([ICUCONFIG_FOR_BUILD], [icu-config], [], [$PATH:/usr/sbin:/sbin])
-        ICU_VERSION_FOR_BUILD=`$ICUCONFIG_FOR_BUILD --noverify --version 2>/dev/null || $ICUCONFIG_FOR_BUILD --version`
+        ICU_VERSION_FOR_BUILD=`$PKG_CONFIG --modversion icu-i18n 2>/dev/null`
         ICU_MAJOR_FOR_BUILD=`echo $ICU_VERSION_FOR_BUILD | cut -d"." -f1`
         ICU_MINOR_FOR_BUILD=`echo $ICU_VERSION_FOR_BUILD | cut -d"." -f2`
         AC_MSG_CHECKING([if MinGW and system versions of ICU are compatible])
-- 
2.11.0

